querybook/webapp/redux/board/reducer.ts

```diff
@@ -49,13 +52,18 @@ export default function (state = initialState, action: BoardAction) {
                 return;
             }
             case '@@board/RECEIVE_BOARD_ITEM': {
-                const { boardItem, boardId } = action.payload;
+                const { boardItem, boardId, itemType, itemId } = action.payload;
                 draft.boardItemById[boardItem.id] = boardItem;
                 if (
                     draft.boardById[boardId].items &&
-                    !(boardItem.id in draft.boardById[boardId].items)
+                    !(boardItem.id in draft.boardById[boardId].items) &&
+                    itemType &&
+                    itemId
                 ) {
                     draft.boardById[boardId].items.push(boardItem.id);
+                    draft.boardById[boardId][itemTypeToKey[itemType]].push(
```

Comment:
this is duplicated information can be computed with selector instead?

Reply:
i think that would be much more complicated. better to have one source of truth. do you think it's necessary/worth it?

Reply:
yeah i was also suggesting it to reduce duplicated information. Do you know where is `boardById[boardId][itemTypeToKey[itemType]]` getting used at?

Reply:
SearchResultItemBoardItemAddButton is using it